pkg/extensions/tbtc/recovery/storage.go

```diff
@@ -0,0 +1,106 @@
+package recovery
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+const (
+	fileName = "derivation_indexes"
+)
+
+type DerivationIndexStorage struct {
+	path string
+}
+
+func NewDerivationIndexStorage(path string) (*DerivationIndexStorage, error) {
```

Comment:
I am thinking one more time about the layout and where `path` should be pointing to and the following structure came to my mind:
```
|
| - current
| - archive
| - snapshot
| - bitcoin <- that's where we are here
```
What do you think?

Reply:
Right now, we have an implementation that looks like:
```
beaushinkle@Beaus-MBP ~/example-storage> tree
.
├── archive
│   ├── 0x0120Db73c97869943Ba8eE87400978051860B6C0
│   │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de
│   ├── 0x03c81f70618702b0fceA6F1656cE07cd4c5132d6
│   │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de
├── current
│   ├── 0x582Db6B4417a7bEB6a2Ac2964824e01d8a5B9548
│   │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de
│   ├── 0x6c672F95DCDA92Dd0aB5F26dBD16eC29deF2312B
│   │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de
└── snapshot
    ├── 0x0120Db73c97869943Ba8eE87400978051860B6C0
    │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de.1613059456465
    ├── 0x03c81f70618702b0fceA6F1656cE07cd4c5132d6
    │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de.1614208084167
    ├── 0x0e0EcbCD48Cb498c5eF47C25fE35A4f86aA72271
    │   └── membership_046dd0d18ac092d4cdaf65198bfa1e3535ae57de.1615579334556
    ├── 0x14A81Fb36A4cd13EaDB1FE271DC522F5CD7B00Fd
└── derivation_indexes
    ├── exte_ublicKey
    └── xpub_zCAk9sk1
        └── 6
```
where `derivation_indexes` lives along side `archive`, `current`, and `snapshot`. This should be enough to make sure that the derivation index stuff never interferes with the existing keep signer serialization stuff and it's backwards compatible.

I was also thinking that if we needed to add more types of persistence, they could have their own unique folder names just like `derivation_indexes`. Do you want to see an additional layer of hierarchy like `bitcoin/derivation_indexes`?

Reply:
> Do you want to see an additional layer of hierarchy like `bitcoin/derivation_indexes`?

Yes. The approach for storing data presented in #744 will let us organize the group data per chain:
```
$data_dir/celo/<current,snapshot,archive>
$data_dir/solana/<current,snapshot,archive>
$data_dir/sovryn/<current,snapshot,archive>
$data_dir/<current,snapshot,archive>     // Ethereum, for backward-compatibility 
                                         // but we can migrate it one day.
```

Having a separate directory for `bitcoin` data would play nicely with this structure, especially if we decide to store more information for `bitcoin` in the future than just derivation indexes.

Reply:
Alrighty: I added a `chainName` constant so that now we get:
```
beaushinkle@Beaus-MBP ~> cd example-storage/
beaushinkle@Beaus-MBP ~/example-storage> tree
.
└── bitcoin
    └── derivation_indexes
        ├── xpub_zCAk9sk1
        │   ├── 172
        │   ├── 3544
        │   ├── 39
        │   ├── 5
        │   ├── 5090
        │   └── 513
        ├── ypub_6Z2Yy4ZH
        │   ├── 1692
        │   ├── 223
        │   ├── 48
        │   └── 8982
        └── zpub_VDaZuQT9
            ├── 112
            ├── 6311
            ├── 6999
            └── 8559
```